pkg/apis/networking/types.go

```diff
@@ -711,3 +707,57 @@ type IPAddressList struct {
 	// Items is the list of IPAddress
 	Items []IPAddress
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ServiceCIDR defines a range of IPs using CIDR format (192.168.0.0/24 or 2001:db2::/64).
+// This range is used by the cluster to allocate the ClusterIPs associated to the Services object.
+type ServiceCIDR struct {
+	metav1.TypeMeta
+	// Standard object's metadata.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#metadata
+	// +optional
+	metav1.ObjectMeta
+	// spec is the desired state of the ServiceCIDR.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	// +optional
+	Spec ServiceCIDRSpec
+	// status represents the current state of the ServiceCIDR.
+	// More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#spec-and-status
+	// +optional
+	Status ServiceCIDRStatus
+}
+
+// ServiceCIDRSpec describe how the ServiceCIDR's specification looks like.
```

Comment:
this is only for Services, ServiceCIDRs and IPAddresses, after all the discussions where we tried to get a more comprehensible definition of the cluster network we settled on , kubernetes only owns the Service CIDRs and the rest is left to implementations, actually we dropped ClusterCIDR because of this